src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1318,61 +1318,94 @@ public override BoundNode VisitConditionalOperator(BoundConditionalOperator node
             VisitCondition(node.Condition);
             var consequenceState = this.StateWhenTrue;
             var alternativeState = this.StateWhenFalse;
+
+            var consequence = VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
+            Unsplit();
+            consequenceState = this.State;
+            var consequenceResult = _result;
+
+            var alternative = VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
+            Unsplit();
+            var alternativeResult = _result;
+
+            bool? getIsNullable(Result result) => result.Type is null ? true : result.Type.IsNullable;
+            bool? resultIsNullable;
             if (IsConstantTrue(node.Condition))
             {
-                VisitConditionalOperand(alternativeState, node.Alternative, isByRef);
-                VisitConditionalOperand(consequenceState, node.Consequence, isByRef);
-                // it may be a boolean state at this point.
+                SetState(consequenceState);
```

Comment:
>SetState(consequenceState); [](start = 16, length = 27)

It used to be that we could stay in a split state in this situation, but it doesn't look like this is the case any more. It feels like we might be sacrificing some accuracy.

Reply:
`PreciseAbstractFlowPass.VisitConditionalOperand` previously set the state to the merged state.

Reply:
> PreciseAbstractFlowPass.VisitConditionalOperand was used previously and that method set the state to the merged state.

I don't see this. What am I missing?

---
In reply to: [174349061](https://github.com/dotnet/roslyn/pull/25359#discussion_r174349061) [](ancestors = 174349061)

Reply:
`PreciseAbstractFlowPass.VisitConditionalOperand` calls `SetState()` so the state is not split.

Reply:
> PreciseAbstractFlowPass.VisitConditionalOperand calls SetState() so the state is not split.

But it sets the state before visiting. Can't visit split the state?

---
In reply to: [174551004](https://github.com/dotnet/roslyn/pull/25359#discussion_r174551004) [](ancestors = 174551004)

Reply:
Fixed.